plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/embedded/EmbeddedEditorModelAccess.java

```diff
@@ -92,6 +110,52 @@ public void updateModel(String prefix, String editablePart, String suffix) {
 		this.viewer.getUndoManager().connect(this.viewer);
 		this.viewer.setRedraw(true);
 	}
+	
+	public void updateModel(final EObject model, final EObject editablePart) {

```

Comment:
maybe just make this updateModel(final String model, final URI editablePartURIFragment)

I'm not quite sure what the benefit is of invoking the serializer inside this methods, because it's just one more call for the caller:

editor.updateModel(serializer.serialize(eObject), uri)

Passing EObjects into this methods may be misleading because the editor doesn't actually reuse them, but creates it's own ResourceSet with a freshly parsed model.